packages/gatsby-plugin-offline/src/gatsby-browser.js

```diff
@@ -1 +1,10 @@
 exports.registerServiceWorker = () => true
+
+exports.onPrefetchPathname = pathData => {
+  if (`serviceWorker` in navigator) {
+    navigator.serviceWorker.controller.postMessage({
+      type: `prefetch`,
+      ...pathData,
```

Comment:
what needs to be done to this data before we can pass on to SW?

Reply:
docs look unclear on what is actually passed, an issue could be opened to fix as well, let me know if we want that!

https://next.gatsbyjs.org/docs/browser-apis/#onPrefetchPathname

Reply:
What needs cached is the code/data files for each path that's prefetched. I can't remember off-hand what is passed to `onPrefetchPathname`. If the actual URLs are passed, there's an internal API on `loader.js` I believe for getting those resource paths.

Importantly though we only need to record prefetching _until_ the SW is loaded as then it'll start caching itself. SW's have lifecycle events which we track here: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/register-service-worker.js

What I think we should do is send those events out via `api-runner-browser` so that then plugins like this one could listen to them. Then you just have a boolean like `swNotLoaded` which defaults to false and while it's false, it queues up page resources and then once the sw loads, it stops listening to prefetching (as the SW is now) and sends all the page resources to the SW to cache them there.

Reply:
>docs look unclear on what is actually passed, an issue could be opened to fix as well, let me know if we want that!

That'd be great too!